androidHyperskillApp/src/main/java/org/hyperskill/app/android/challenge/ui/ChallengeCard.kt

```diff
@@ -0,0 +1,77 @@
+package org.hyperskill.app.android.challenge.ui
+
+import androidx.compose.foundation.layout.aspectRatio
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import org.hyperskill.app.android.core.view.ui.widget.compose.ShimmerLoading
+import org.hyperskill.app.android.core.view.ui.widget.compose.WidgetDataLoadingError
+import org.hyperskill.app.challenges.widget.presentation.ChallengeWidgetFeature.Message
+import org.hyperskill.app.challenges.widget.view.model.ChallengeWidgetViewState
+
+@Composable
+fun ChallengeCard(
+    viewState: ChallengeWidgetViewState,
+    onNewMessage: (Message) -> Unit
+) {
+    when (viewState) {
+        ChallengeWidgetViewState.Idle, ChallengeWidgetViewState.Empty -> {
+            // no op
+        }
+        is ChallengeWidgetViewState.Loading -> {
+            ShimmerLoading(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .aspectRatio(ratio = 0.5f)
+            )
+        }
+        ChallengeWidgetViewState.Error -> {
+            WidgetDataLoadingError(
```

Comment:
If `WidgetDataLoadingError` supports injection of the error messages, it would be nice to use `challenge_widget_network_error_text`